docs/rst/tutorials/core/qos/mutable/mutable.rst

```diff
@@ -0,0 +1,296 @@
+
+The ROS 2 middleware layer (see :ref:`ROS 2 RMW <RO2Vendors>`) provides APIs to get handles to the objects of the inner DDS implementation, which is needed to be able to change the mutable Qos.
+Thus, this tutorial will also show how to use that powerful tool.
+For another demo on how to access inner RMW entities, see `demo_nodes_cpp_native <https://github.com/ros2/demos/tree/master/demo_nodes_cpp_native>`_.
+
+Prerequisites
+-------------
+
+The first prerequisite is to have Vulcanexus Humble installed (see :ref:`Linux binary installation <linux_binary_installation>` or :ref:`Linux installation from sources <linux_source_installation>`).
+
+Also, before starting this tutorial, user should be familiar with creating a workspace and creating a package, as well as familiar with parameters and their function in a ROS 2 system.
+The recommendation is to first complete the following tutorials:
+
+    * :ref:`Modifying Ownership and Ownership Strength QoS Policy <tutorials_qos_ownership_ownership>`
+
+    * :ref:`Writing a simple publisher and subscriber (C++) <CppPubSub>`
+
+    * :ref:`Understanding parameters <ROS2Params>`
+
+    * :ref:`Using parameters in a class (C++) <CppParamNode>`
+
+    * :ref:`Monitoring for parameter changes (C++) <MonitorParams>`
+
+Create a package
+----------------
+
+First, open a new terminal and source your Vulcanexus installation.
+
+.. code-block:: bash
+
+    source /opt/vulcanexus/humble/setup.bash
+
+
+Next lines will create a package in a new workspace folder named `vulcanexus_ws` in `~/`.
+
+.. code-block:: bash
+
+    mkdir ~/vulcanexus_ws && mkdir ~/vulcanexus_ws/src && cd ~/vulcanexus_ws/src
+    ros2 pkg create --build-type ament_cmake \
+        vulcanexus_change_mutable_qos --dependencies rclcpp
+
+
+Make sure to add the description, maintainer email and name, and license information to the `package.xml` file of the created package, as explained in the aforementioned tutorials.
+
+Write sources
+-------------
+
+First, the source file of the first publisher will be created, the Publisher 1.
+Inside the `vulcanexus_ws/src/vulcanexus_change_mutable_qos/src` directory, create a new file called `change_mutable_qos_pub1.cpp` and paste the following code within:
+
+.. literalinclude:: /resources/tutorials/core/qos/mutable/vulcanexus_change_mutable_qos/src/change_mutable_qos_pub1.cpp
+    :language: c++
+
+
+The code for the Publisher 2 is the same, just changing the Publisher 1 names for Publisher 2 names.
+Inside the `vulcanexus_ws/src/vulcanexus_change_mutable_qos/src` directory, create a new file called `change_mutable_qos_pub2.cpp` and paste the following code within:
+
+.. literalinclude:: /resources/tutorials/core/qos/mutable/vulcanexus_change_mutable_qos/src/change_mutable_qos_pub2.cpp
+    :language: c++
```

Comment:
I think we should unify these 2 files into a single class that takes the node name and initial strength as constructor parameters. Then, the main can just take those 2 arguments and create the class accordingly

Reply:
Done